src/CoreFn/Ident.purs

```diff
@@ -23,16 +18,15 @@ data Ident
   -- A generated name for an identifier
   --
   | GenIdent (Maybe String) Int
+  -- |
+  -- A generated name used only for type-checking
+  --
+  | UnusedIdent
```

Comment:
I'm not sure whether to keep this or not. There's nothing that can be code-generated from it, but the case needs to be handled or `unsafePartial` used.